src/StructuredLogger.Tests/Analyzers/ProjectReferenceGraphTests.cs

```diff
@@ -0,0 +1,442 @@
+// using System;
+// using System.Collections.Generic;
+// using System.IO;
+// using System.Linq;
+// using Moq;
+// using Xunit;
+// 
+// namespace Microsoft.Build.Logging.StructuredLogger.UnitTests
+// {
+//     /// <summary>
+//     /// Unit tests for the <see cref="Microsoft.Build.Logging.StructuredLogger.ProjectReferenceGraph"/> class.
+//     /// </summary>
+//     public class ProjectReferenceGraphTests
+//     {
+//         /// <summary>
+//         /// Tests that the constructor does not throw when the EvaluationFolder contains a ProjectEvaluation with no Items folder.
+//         /// </summary>
+// //         [Fact] [Error] (30-36)CS0117 'Record' does not contain a definition for 'Exception'
+// //         public void Constructor_WithMissingItemsFolder_DoesNotThrow()
+// //         {
+// //             // Arrange
+// //             var build = new DummyBuild();
+// //             var evalFolder = new DummyFolder("EvaluationFolder");
+// //             var projectEval = new DummyProjectEvaluation("C:/Project/Test.csproj");
+// //             // Do not add Items folder to project evaluation.
+// //             evalFolder.AddChild(projectEval);
+// //             build.EvaluationFolder = evalFolder;
+// // 
+// //             // Act & Assert
+// //             var exception = Record.Exception(() => new Microsoft.Build.Logging.StructuredLogger.ProjectReferenceGraph(build));

```

Comment:
The Record here conflicted between `Xunit` and `Microsoft.Build.Logging` namespaces (later got the precedence)
But this would be hard to catch